src/_pytest/pathlib.py

```diff
@@ -365,3 +365,105 @@ def fnmatch_ex(pattern: str, path) -> bool:
 def parts(s: str) -> Set[str]:
     parts = s.split(sep)
     return {sep.join(parts[: i + 1]) or sep for i in range(len(parts))}
+
+
+def import_module(path, modname=None, ensuresyspath=True):
+    """ return path as an imported python module.
+
+    If modname is None, look for the containing package
+    and construct an according module name.
+    The module will be put/looked up in sys.modules.
+    if ensuresyspath is True then the root dir for importing
+    the file (taking __init__.py files into account) will
+    be prepended to sys.path if it isn't there already.
+    If ensuresyspath=="append" the root dir will be appended
+    if it isn't already contained in sys.path.
+    if ensuresyspath is False no modification of syspath happens.
+
+    Special value of ensuresyspath=="importlib" is intended
+    purely for using in pytest, it is capable only of importing
+    separate .py files outside packages, e.g. for test suite
+    without any __init__.py file. It effectively allows having
+    same-named test modules in different places and offers
+    mild opt-in via this option. Note that it works only in
+    recent versions of python.
+    """
+    import py.error
```

Comment:
How hard would it be to avoid `py.error` here? (Trying to reduce the `py` modules we directly depend on).